sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -4882,3 +4882,65 @@ def sdc_pandas_series_skew_impl(self, axis=None, skipna=None, level=None, numeri
         return numpy_like.skew(self._data)
 
     return sdc_pandas_series_skew_impl
+
+
+@sdc_overload_method(SeriesType, 'combine')
+def sdc_pandas_series_combine(self, other, func, fill_value=None):
+    """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+
+    Pandas API: pandas.Series.combine
+
+    Limitations
+    -----------
+    - Only supports the case when data in series of the same type
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/series/series_combine.py
+       :language: python
+       :lines: 27-
+       :caption: Combined the Series with a Series according to func.
+       :name: ex_series_combine
+
+    .. command-output:: python ./series/series_combine.py
+       :cwd: ../../../examples
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas Series method :meth:`pandas.Series.combine` implementation.
+
+    .. only:: developer
+
+       Tests: python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_combine*
+    """
+    _func_name = 'Method Series.combine().'
+
+    ty_checker = TypeChecker(_func_name)
+    ty_checker.check(self, SeriesType)
+
+    ty_checker.check(other, SeriesType)
+
+    if not isinstance(fill_value, (types.Omitted, types.NoneType, types.Number)) and fill_value is not None:
+        ty_checker.raise_exc(fill_value, 'number', 'fill_value')
+
+    def sdc_pandas_series_combine_impl(self, other, func, fill_value=None):
+
+        if fill_value is None:
+            fill_value = numpy.nan
```

Comment:
This will make fill_value type undefined at compile time. You can probably use the same approach as in operators: https://github.com/IntelPython/sdc/blob/e87095a53f4810e26f944288a51dbe7dbdc2c5c0/sdc/sdc_function_templates.py#L144